reconcile/utils/oc.py

```diff
@@ -439,6 +440,26 @@ def get_service_account_username(user):
         name = user.split('/')[1]
         return "system:serviceaccount:{}:{}".format(namespace, name)
 
+    def recycle_orphan_pods(self, dry_run, namespace, resource):
+        pods = self.get(namespace, 'Pod')['items']
+        for p in pods:
+            owner = self.get_obj_root_owner(namespace, p)
+            if resource.kind != owner['kind']:
+                continue
+            if resource.name != owner['metadata']['name']:
+                continue
+            # we found a pod we want to recycle
+            name = p['metadata']['name']
+            self.delete(namespace, 'Pod', name)
+            self.validate_pod_ready(namespace, name)
+
+    @retry(max_attempts=20)
+    def validate_pod_ready(self, namespace, name):
```

Comment:
Maybe 
```suggestion
    def validate_recycled_statefuset_pod_ready(self, namespace, name):
```
Sorry for the long names, I must have been a java developer in a previous life

Reply:
this is a generic function, the pod just has to become ready

Reply:
Hmmm, ok, you're right. This can be indeed a function that doesn't have to do with stateful sets